base/dict.jl

```diff
@@ -189,6 +189,13 @@ function grow_to!{K,V}(dest::Associative{K,V}, itr, st)
     return dest
 end
 
+
+"""
+    similar(dictionary)
+
+Create an empty dictionary with the same element types for key and value of the given source
```

Comment:
Much better.
"same key and value types as the given source" or "same key and value types as those of the given source"?